sessions/2018/november/session-18.md

```diff
@@ -0,0 +1,198 @@
+# SOFTWARE ENGINEERING IN JAVA
+
+## Session 18 (27/11/2018)
+
+- Reviewing homework
+- Adding BugSnag to my API code
+- Partial Git commits
+
+### Notes
+
+#### Reviewing my Homework
+
+Roger was reviewing my homework and made me several suggestions that has to be addressed:
+
+- If I go to <http://localhost:8080/api/factorial/30>, I receive the result `265252859812191058636308480000000`. But If use the frontend this is rounding the result to `2.6525285981219107e+32`. So this must also show the numbers with the same precision as in the backend.
+
+- Verify numbers are being showed as numbers and not as strings.
+
+Also he talk to me about source maps see [here][2] and [here][3] to minimize included JS or CSS files in order to the web application loads faster.
+
+#### Adding BugSnag to my API code
+
+To add BugSnag to my code I had to follow the options described in their oficial repositories to [integrate with Java][1]. Then create an account into their website. I did under my particular name (also worked as my company name). After that, I had to create a new project specifying the programming language for which they provided to me the instructions to complete and test the integration.
+
+Basically the configurations are the following:
+
+At the controller, I injected the BugSnag class and throwed a test exception:
+
+```java
+public class CalculatorController {
+
+    private final Calculator calculator;
+    private final Bugsnag bugsnag;
+
+    public CalculatorController(Calculator calculator, Bugsnag bugsnag) {
+        this.calculator = calculator;
+        this.bugsnag = bugsnag;
+    }
+
+    @GetMapping("/fibonacci/{number}")
+    public List<BigInteger> getFibonacci(@PathVariable("number") Integer number){
+      bugsnag.notify(new RuntimeException("Test error"));
+      return calculator.getFibonacci(number);
+    }
+...
+}
+```
+
+And I just copied the BugsnagConfig class code snippet into my IDE and the class was automatically created.  
+
+After this was done, running my application and testing that map, I could to visit my BugSnag desktop and see the exception being catched at:
```

Comment:
The exception is not caught permanently by Bugsnag - it will still 'bubble up', so if there's another `catch` block above where Bugsnag catches the exception it will still be caught there.

```suggestion
After this was done, running my application and testing that map, I could to visit my BugSnag desktop and see the exception being reported at:
```